### PR TITLE
refactor: improve cli

### PR DESCRIPTION
**Pull request by aarcex3**
# Code proposal

The main cli has been refactor and a new command has been added, namely `generate`.
This is a proposal to generate a folder like the ones on the fullstack [example](https://github.com/litestar-org/litestar-fullstack/tree/937d441d920ad5952f2178ec8dfe1f90f5492b5e/src/app/domain), where a resource consists of a controller, service, dtos, models and a repository file.
The proposal is to follow a Nestjs like philosophy when it comes to the cli.

# Related Issue
None

### CODE REVIEW

File: .vscode/settings.json
Comment (ch-iv) [inline .vscode/settings.json:1]: Remove IDE configuration files from the commit.

File: README.md [hunk 1]
```diff
@@ -7,13 +7,13 @@ project using the Litestar CLI.
...
 
-```
+```bash
 litestar project init --app-name MyProject
 ```
 
 This command is used to initialize a Litestar project named MyProject in the current working directory. MyProject will have the following tree structure:
 
-```
+```bash
```
Comment (ch-iv) [inline README.md:16]: Is there a specific reason behind this change?

File: pyproject.toml [hunk 2]
```diff
@@ -306,6 +306,7 @@ distribution = true
 [tool.pdm.dev-dependencies]
 dev = [
     "litestar[standard,cryptography,jinja]>=2.9.1",
+    "black>=24.8.0",
```
Comment (ch-iv) [inline pyproject.toml]: The project uses ruff for linting. Remove black dependency and lint the PR using ruff.

File: src/litestar_manage/cli.py [hunk 3]
```diff
@@ -10,12 +10,17 @@
...
-def project_group() -> None:
-    """Manage Scaffolding Tasks."""
+def is_project_initialized() -> bool:
+    output_dir = Path.cwd()
+    return (output_dir / "src").exists() or (output_dir / "venv").exists()
+
 
+@click.group(cls=LitestarGroup)
+def cli():
+    pass
```
Comment (ch-iv) [inline src/litestar_manage/cli.py]: This is meant to be an extension of the LItestar's CLI, so it doesn't really make sense for the group to be named CLI. Rename this group back to `project`.

File: src/litestar_manage/cli.py [hunk 4]
```diff
@@ -31,14 +36,38 @@ def project_group() -> None:
...
 
+    if is_project_initialized():
+        click.echo("Project already initialized.")
+        return
+
+    ctx = RenderingContext(app_name=app_name)
     render_template(template_dir, output_dir, ctx, run_ruff=True)
 
     packages_to_install = ["litestar"]
-    venv_name = "venv"
```
Comment (ch-iv) [inline src/litestar_manage/cli.py:41]: We don't want to make assumptions about what the virtual environment name is, nor do we want to hard code that assumption. Factor out the virtual environment name into a constant and possibly make a list of possible venv names to check against.

**Review by ch-iv — CHANGES_REQUESTED** [review]
Hey! Thanks for the pull request, your work is really appreciated. I totally agree with having an option to generate individual resources. In addition to some of the comments that I've left, I also think that you should not use `RenderingContext` for rendering a resource. Instead create another context class (something like `ResourceRenderingContext` with resource specific context fields) and use it to render resources instead.

File: README.md (hunk 1 above)
Comment (aarcex3) [reply README.md:16]: It looks cleaner in my opinion, I can remove it if you want

**Comment by aarcex3** [comment]
> Hey! Thanks for the pull request, your work is really appreciated. I totally agree with having an option to generate individual resources. In addition to some of the comments that I've left, I also think that you should not use `RenderingContext` for rendering a resource. Instead create another context class (something like `ResourceRenderingContext` with resource specific context fields) and use it to render resources instead.

Would it be ok to make a subclass of RenderingContext for this matter? Or maybe create an abstract RenderingContext from which the other can inherent? Because in most cases we just render jinja files (and others), right?

**Comment by ch-iv** [comment]
Since `RenderingContext` is a dataclass it wouldn't really make sense to have another class inherit from it. `RenderingContext` just a structure holding some values used during rendering. I would just create another dataclass called `ResourceRenderingContext` without any inheritance. And maybe rename `RenderingContext` to `AppRenderingContext`.

**Comment by ch-iv** [comment]
@aarcex3 Thank you for your efforts.

**Pull request merged**